shell/platform/darwin/macos/README.md

```diff
@@ -0,0 +1,48 @@
+# Flutter macOS Embedder
+
+This directory contains files specific to the Flutter macOS embedder. The
+embedder is built as a macOS framework that is linked into the target
+application. The framework is composed both of macOS-specific code and code
+shared with the iOS embedder. These can be found in:
+```
+flutter/shell/platform/darwin/common/framework
+flutter/shell/platform/darwin/macos/framework
+```
+
+Additionally, the framework relies on utility code shared across all embedders,
+which is found in:
+```
+flutter/shell/platform/common
+```
+
```

Comment:
Consider linking back to wiki for more background info:

```suggestion
To learn more, see the [Engine architecture wiki](https://github.com/flutter/flutter/wiki/The-Engine-architecture).

```

Reply:
Nice! Done.